src/main/java/com/flippingutilities/ui/combinationflips/CombinationFlipCreationPanel.java

```diff
@@ -0,0 +1,572 @@
+        Map<Integer, Integer> targetValues = getTargetValuesForMaxRecipeCount(itemIdToPartialOffers);
+
+        idToHeader.keySet().forEach(id -> {
+            int targetValue = targetValues.get(id);
+            List<PartialOffer> partialOffers = itemIdToPartialOffers.get(id);
+            CombinationItemHeaderPanel combinationItemHeaderPanel = idToHeader.get(id);
+            combinationItemHeaderPanel.setTargetValueDisplay(targetValue);
+            bodyPanel.add(createOffersPanel(partialOffers, combinationItemHeaderPanel, targetValue));
+            if (id == parentOffer.getItemId()) {
+                //empty panel occupies the space under the arrow
+                JPanel emptyPanel = new JPanel();
+                emptyPanel.setBackground(Color.BLACK);
+                bodyPanel.add(emptyPanel);
+            }
+        });
+
+        handleItemsHittingTargetConsumptionValues();
+    }
+
+    /**
+     * This method computes the initial target values for each of the items when the panel
+     * first shows up. This is so that the user doesn't have to manually input them (tho they can still
+     * adjust them if they want). The target values are selected such that the max amount of
+     * combinations/recipes can be made. Here is an example:
+     *
+     * Lets say there is a recipe where for every item C you need 3 of item A and 5 of item B. We
+     * can shorten it by saying the recipe is 1C, 3A, and 5B. Now, lets say we have a quantity of 5 for item C,
+     * quantity of 9 for item A and a quantity of 25 for item B. What is the max amount of this recipe you can make?
+     * The max amount is only 3 because it is limited by that fact you only have 9 A items, even though the amount
+     * of C and B items you have can support more recipes.
+     *
+     * Once we know the max amount of recipes the offers can support, we can multiply the max amount by the
+     * amount the recipe needs of each item to know how much of each item we will need to make the max amount of
+     * recipes. Continuing with the example above, our max recipes is 3. To get the amount of C items, we would do
+     * 1 * 3. To get the amount of B items, we would do 5 * 3. To get the amount of A items, we would do 3 * 3.
+     *
+     * In short:
+     * recipe = 3A, 5B, 1C
+     * quantities = 9A 25B 5C
+     * max # of this recipe supported per item: A=3, B=5, C=5
+     * # of recipe supported in actuality: 3, cause A is constraining it.
+     * target values: A = 9(3 * 3), B = 15(3 * 5), C = 3(3 * 1)
+     *
+     * @param itemIdToPartialOffers all the suitable partial offers for each item
+     */
+    private Map<Integer, Integer> getTargetValuesForMaxRecipeCount(
+            Map<Integer, List<PartialOffer>> itemIdToPartialOffers) {
+        Map<Integer, Integer> itemIdToQuantity = recipe.getItemIdToQuantity();
+
+        Map<Integer, Integer> itemIdToMaxRecipesThatCanBeMade = getItemIdToMaxRecipesThatCanBeMade(itemIdToPartialOffers);
```

Comment:
That method is used in one other place to see if any items cannot fulfill even one of that recipe, which means I need access to all the item's amounts, not just the highest one. However, you are right in that I don't actually need the item id, as I just use the values. But, I don't see the point in changing it (to a list for example) as the map is more useful (tho we don't use that benefit currently) via relating the quantities to each item and is not harder to construct than a list.

Reply:
I think not changing `getItemIdToMaxRecipesThatCanBeMade` is fine, but perhaps mapping the result to an array of the values might be slightly better for the current context?
Seems like this would simplify the flow a bit, but it's a nitpick at this point so it's not a big deal